internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -226,9 +226,10 @@ func (opts *Opts) createResources() error {
 
 	if err := opts.createCluster(); err != nil {
 		var target *atlas.ErrorResponse
-		if errors.As(err, &target) && target.ErrorCode == "CANNOT_CREATE_FREE_CLUSTER_VIA_PUBLIC_API" {
```

Comment:
we need to do a string comparison on the reasoning here. we've been hiding one different error because of this assumption